uniffi/src/interface/mod.rs

```diff
@@ -451,12 +458,16 @@ impl APIConverter<Argument> for weedle::argument::Argument<'_> {
 
 impl APIConverter<Argument> for weedle::argument::SingleArgument<'_> {
     fn convert(&self, ci: &ComponentInterface) -> Result<Argument> {
-        if self.attributes.is_some() {
-            bail!("argument attributes are not supported yet");
-        }
         Ok(Argument {
             name: self.identifier.0.to_string(),
             type_: (&self.type_).resolve_type_definition(ci)?,
+            by_ref: match &self.attributes {
+                None => false,
+                Some(def) => def.body.list.iter().find(| attr| match attr {
+                    weedle::attribute::ExtendedAttribute::NoArgs(attr) => (attr.0).0 == "ByRef",
+                    _ => false
+                }).is_some()
+            },
```

Comment:
@tarikeshaq you've been working with attributes a bit over in https://github.com/mozilla/uniffi-rs/pull/31, is there anything from that PR that's worth extracting and landing so it can be re-used here?

Reply:
Yup! Opened #187 for that!